src/features/import-map.js

```diff
@@ -37,23 +33,23 @@ export function mergeImportMap(originalMap, newMap) {
   return originalMap;
 }
 
-systemJSPrototype.resolve = function (id, parentUrl) {
-  parentUrl = parentUrl || baseUrl;
-
+systemJSPrototype.prepareImport = function () {
   if (acquiringImportMaps) {
     acquiringImportMaps = false;
-    Array.prototype.forEach.call(document.querySelectorAll('script[type="systemjs-importmap"]'), function (script) {
-      importMapPromise = importMapPromise.then(function (map) {
-        return (script._j || script.src && fetch(script.src).then(function (resp) {return resp.json()}) || Promise.resolve(JSON.parse(script.innerHTML)))
-        .then(function (json) {
-          return mergeImportMap(map, parseImportMap(json, script.src || baseUrl));
-        });
+    return Promise.all(Array.prototype.map.call(document.querySelectorAll('script[type="systemjs-importmap"]'), function (script) {
+      return (script._j || script.src && fetch(script.src).then(function (resp) {return resp.json()}) || Promise.resolve(JSON.parse(script.innerHTML)))
+      .then(function (json) {
+        return mergeImportMap(finalMap || baseMap, parseImportMap(json, script.src || baseUrl));
```

Comment:
This should just be `finalMap`?

Reply:
Also `mergeImportMap` is synchronous and applies the map to the base map so this can just be - 

```suggestion
        mergeImportMap(finalMap, parseImportMap(json, script.src || baseUrl));
      });
```